File: LeetCode/739-Daily-Temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n=temperatures.size();
        stack<pair<int, int>> s;
        vector<int> ans(n, 0);

        for(int i=n-1; i>=0; i--) {
            while(!s.empty() && temperatures[i]>=s.top().first)
                s.pop();
            if(!s.empty()) ans[i]=s.top().second-i;
            s.push({temperatures[i], i});
        }

        return ans;
    }
};

File: LeetCode/739-Daily-Temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        ans = [0]*len(temperatures)
        stack = []

        for i, t in enumerate(temperatures):
            while stack and t>stack[-1][0]:
                a, b = stack.pop()
                ans[b] = i-b
            stack.append([t, i])

        return ans